### DISCUSSION TITLE
DevKit & Plugin Documentation

### DISCUSSION THREAD

**Discussion by Brian-McBride** [nrwl/nx#11132] [Feature Requests]
I've asked about documentation on developing plugins in the past, and I've received the answer, "look at other plugins."

Nx is great - but I feel there is a lost opportunity here by not having great plugin documentation. I've been trying to reverse engineer some plugins to make a nice GraphQL app and lib generators. However, it is time-consuming and frustrating to have to use existing code as documentation - when so many of the Nx plugins don't actually follow their own standard nor match what the Plugin generator creates. As a simple example, Nx uses `collection.json` instead of `generators.json`. 

There are also a lot of utilities that live in the Nx repo that are not part of the DevKit plugin, but almost all Nx plugins use in their own codebase. If I want to use Nx code as a template, I also have to go and replicate all these utils (even if it is copy/paste, it is still more finding, reverse engineering, and then moving it into my own repo).

I don't mean to complain here. I appreciate all that Nrwl has given the community. I imagine taking on Lerna also reduces the amount of time that can be put on Nx. It seems that by making plugins easier to build - we will get more community members adding to the available library, which will then bring in more Nx users in general.

**Comment by JKKGBE** [comment]
I too wish someone would consider this issue.

**Reply by patdx** [reply]
It is probably complicated by the fact that Nx schematics are based on [Angular schematics](https://angular.io/guide/schematics-for-libraries) and have some backwards compatibility with them. It is probably easier to grasp for developers coming from an Angular background. (Though, personally I still find it confusing. I can't remember which term is the Nx term vs Angular term)
